### PR TITLE
Fix for cognitive services API version

### PR DESCRIPTION
**Pull request by Dag-Calafell-MCA**
### Motivation and Context

Fix for issue #425 

The older API version was no longer working.  It allows for deploying a new Azure OpenAI resource.

### Description

Updates the API version and SKU of the Azure OpenAI 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by Dag-Calafell-MCA** [comment]
@microsoft-github-policy-service agree

**Review by crickman — DISMISSED** [review]

**Review by glahaye — APPROVED** [review]

File: scripts/deploy/main.bicep [hunk 1]
```diff
@@ -78,7 +78,7 @@ param deploySpeechServices bool = true
 @description('Whether to deploy the web searcher plugin, which requires a Bing resource')
 param deployWebSearcherPlugin bool = false
 
-@description('Whether to deploy pre-built binary packages to the cloud')
+@description('Whether to deploy binary packages to the cloud')
```
Comment (glahaye) [inline scripts/deploy/main.bicep:81]:
```suggestion
@description('Whether to deploy pre-built binary packages to the cloud')
```

File: scripts/deploy/main.bicep [hunk 2]
```diff
@@ -385,18 +387,6 @@ resource appServiceWebConfig 'Microsoft.Web/sites/config@2022-09-01' = {
           name: 'SemanticMemory:Services:AzureOpenAIEmbedding:Deployment'
           value: embeddingModel
         }
```
Comment (glahaye) [inline scripts/deploy/main.bicep:389]:
```suggestion
        }
        {
          name: 'SemanticMemory:Services:OpenAI:TextModel'
          value: completionModel
        }
        {
          name: 'SemanticMemory:Services:OpenAI:EmbeddingModel'
          value: embeddingModel
        }
        {
          name: 'SemanticMemory:Services:OpenAI:APIKey'
          value: aiApiKey
        }
```

File: scripts/deploy/main.bicep [hunk 3]
```diff
@@ -580,18 +582,6 @@ resource appServiceMemoryPipelineConfig 'Microsoft.Web/sites/config@2022-09-01'
         name: 'SemanticMemory:Services:AzureFormRecognizer:APIKey'
         value: ocrAccount.listKeys().key1
       }
```
Comment (glahaye) [inline scripts/deploy/main.bicep:584]: {
        name: 'SemanticMemory:Services:OpenAI:TextModel'
        value: completionModel
      }
      {
        name: 'SemanticMemory:Services:OpenAI:EmbeddingModel'
        value: embeddingModel
      }
      {
        name: 'SemanticMemory:Services:OpenAI:APIKey'
        value: aiApiKey
      }

Comment (glahaye) [inline scripts/deploy/main.bicep:584]:
```suggestion
      }
      {
        name: 'SemanticMemory:Services:OpenAI:TextModel'
        value: completionModel
      }
      {
        name: 'SemanticMemory:Services:OpenAI:EmbeddingModel'
        value: embeddingModel
      }
      {
        name: 'SemanticMemory:Services:OpenAI:APIKey'
        value: aiApiKey
      }
```

**Review by glahaye — APPROVED** [review]

**Review by dehoward — APPROVED** [review]

**Pull request merged**